src/build/specialize_types/IR.zig

```diff
@@ -125,129 +147,157 @@ pub const Expr = union(enum) {
 
     compiler_bug: Problem.Compiler,
 
-    /// todo
+    /// List of Expr
     pub const List = collections.SafeList(@This());
-    /// todo
+    /// Index into an Expr.List
     pub const Idx = List.Idx;
-    /// todo
+    /// Sublist reference of Expr.List
     pub const Slice = List.Slice;
-    /// todo
+    /// Sublist reference into Expr.List that is guaranteed to be non-empty
     pub const NonEmptySlice = List.NonEmptySlice;
-    /// todo
+    /// ??? Type annotated expression e.g. `1: U64` ?
     pub const Typed = struct {
         expr: Expr.Idx,
         type: Type.Idx,
-        /// todo
+        /// List of Expr.Typed
         pub const List = collections.SafeMultiList(@This());
-        /// todo
+        /// Sublist reference into Expr.Typed.List
         pub const Slice = Typed.List.Slice;
     };
 };
 
 /// A definition, e.g. `x = foo`
 pub const Def = struct {
+    /// ??? Is pattern for what's to the left of `=`?
     pattern: Pattern.Idx,
-    /// Named variables in the pattern, e.g. `a` in `Ok a ->`
+    /// Named variables in the pattern, e.g. `a` in `Ok a ->` ??? A when branch is not a Def right?
```

Comment:
You're right, a better example would be "e.g. `a` and `b` in { a, b } = some_record`"